recipes/tls-securepair-to-tlssocket/src/workflow.ts

```diff
@@ -0,0 +1,114 @@
+import { getNodeImportStatements, getNodeImportCalls as getNodeNodeImportCall } from "../../../utils/src/ast-grep/import-statement.ts";
+import { getNodeRequireCalls } from "../../../utils/src/ast-grep/require-call.ts";
+import { updateBinding } from "../../../utils/src/ast-grep/update-binding.ts";
+import { removeLines } from "../../../utils/src/ast-grep/remove-lines.ts";
+import type { Edit, SgRoot, Range, SgNode } from "@codemod.com/jssg-types/main";
+import type Js from "@codemod.com/jssg-types/langs/javascript";
+
+function getClosest(node: SgNode<Js>, kinds: string[]): SgNode<Js> | null {
+  let current = node.parent();
+ 
+  while (current) {
+    if (kinds.includes(current.kind())) return current;
+    current = current.parent();
+  }
+ 
+  return null;
+}
+
+export default function transform(root: SgRoot<Js>): string | null {
+  const rootNode = root.root();
+  const edits: Edit[] = [];
+  const linesToRemove: Range[] = [];
+
+  const importNodes = [
+    ...getNodeImportStatements(root, "tls"),
+    ...getNodeRequireCalls(root, "tls"),
+    ...getNodeNodeImportCall(root, "tls")
+  ];
+
+  for (const node of importNodes) {
+    const change = updateBinding(node, { old: "SecurePair", new: "TLSSocket" });
+    if (change?.edit) edits.push(change.edit);
+    if (change?.lineToRemove) linesToRemove.push(change.lineToRemove);
+  }
+
+  const newExpressions = rootNode.findAll({
+    rule: {
+      kind: "new_expression",
+      has: {
+        any: [
+          { kind: "member_expression", has: { field: "property", regex: "^SecurePair$" } },
+          { kind: "identifier", regex: "^SecurePair$" }
+        ]
+      }
+    }
+  });
+
+  for (const node of newExpressions) {
+    const callee = node.field("constructor");
+    if (!callee) continue;
+
+    let newConstructorName = "TLSSocket";
+    if (callee.kind() === "member_expression") {
+        const object = callee.field("object");
+        if (object) {
+            newConstructorName = `${object.text()}.TLSSocket`;
+        }
+    }
+
+    edits.push(node.replace(`new ${newConstructorName}(socket)`));
+    const declarator = getClosest(node, ["variable_declarator"]);
+    if (declarator) {
+      const idNode = declarator.field("name");
+      if (idNode) {
+        const oldName = idNode.text();
+        let newName = "socket";
+        if (oldName !== "pair" && oldName !== "SecurePair") {
+          if (oldName.includes("Pair")) newName = oldName.replace("Pair", "Socket");
+          else if (oldName.includes("pair")) newName = oldName.replace("pair", "socket");
+        }
```

Comment:
This code block feels a little odd to me, could you explain the idea behind it? It seems very dependent on `oldName`, which feels a bit tricky. Why do we need it? Maybe we could do this in another way?